server/controllers/finance/reports/shared.js

```diff
@@ -0,0 +1,85 @@
+  { field : 'invoice_uuid', displayName : 'FORM.LABELS.INVOICE' },
+  { field : 'service_id', displayName : 'FORM.LABELS.SERVICE' },
+  { field : 'inventory_uuid', displayName : 'FORM.LABELS.INVENTORY' },
+
+  { field : 'flux_id', displayName: 'STOCK.FLUX' },
+  { field : 'status', displayName: 'STOCK.STATUS.LABEL' },
+  { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
+  { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
+
+  { field : 'entry_date_from', displayName: 'STOCK.ENTRY_DATE', comparitor: '>', isDate : true },
+  { field : 'entry_date_to', displayName: 'STOCK.ENTRY_DATE', comparitor: '<', isDate : true },
+  { field : 'expiration_date_from', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '>', isDate : true },
+  { field : 'expiration_date_to', displayName: 'STOCK.EXPIRATION_DATE', comparitor: '<', isDate : true },
+  { field : 'is_exit', displayName : 'STOCK.OUTPUT' },
+  { field : 'label', displayName : 'STOCK.LOT' },
+  { field : 'depot_uuid', displayName : 'STOCK.DEPOT' },
+
+  { field : 'description', displayName : 'FORM.LABELS.DESCRIPTION' },
+  { field : 'entity_uuid', displayName : 'FORM.LABELS.ENTITY' },
+
+  { field : 'display_name', displayName : 'FORM.LABELS.NAME' },
+  { field : 'sex', displayName : 'FORM.LABELS.GENDER' },
+  { field : 'code', displayName : 'FORM.LABELS.CODE' },
+  { field : 'dateBirthFrom', displayName : 'FORM.LABELS.DOB', comparitor : '>', isDate : true },
+  { field : 'dateBirthTo', displayName : 'FORM.LABELS.DOB', comparitor : '<', isDate : true },
+  { field : 'dateEmbaucheFrom', displayName : 'FORM.LABELS.DATE_EMBAUCHE', comparitor : '>', isDate : true },
+  { field : 'dateEmbaucheTo', displayName : 'FORM.LABELS.DATE_EMBAUCHE', comparitor : '<', isDate : true },
+  { field : 'grade_id', displayName : 'FORM.LABELS.GRADE' },
+  { field : 'fonction_id', displayName : 'FORM.LABELS.FUNCTION' },
+  { field : 'service_id', displayName : 'FORM.LABELS.SERVICE' },
+  { field : 'type_ids', displayName : 'FORM.LABELS.TRANSACTION_TYPE' },
+
+  { field : 'reference', displayName : 'FORM.LABELS.REFERENCE' },
+  { field : 'reversed', displayName : 'CASH.REGISTRY.REVERSED_RECORDS' },
+  { field : 'limit', displayName : 'FORM.LABELS.LIMIT' },
+  { field : 'period', displayName : 'TABLE.COLUMNS.PERIOD', isPeriod : true },
+  { field : 'custom_period_start', displayName : 'PERIODS.START', isDate : true, comparitor : '>' },
+  { field : 'custom_period_end', displayName : 'PERIODS.END', isDate : true, comparitor : '<' },
+];
+
+_.sortBy(columns, c => c.field);
+
+// translation key mappings for dynamic filters
+// Basically, to show a pretty filter bar, this will translate URL query string
+// into human-readable text to be placed in the report, showing the properties
+// filtered on.
+function formatFilters(qs) {
+  return columns.filter(column => {
+    const value = qs[column.field];
+
+    if (!_.isUndefined(value)) {
+      if (column.isPeriod) {
```

Comment:
It will be better the handler service situation also. I am seeing something like ```Service : 2``` on the report filter. can you replace it with the service label?

Reply:
We could do that if we had the correct translation for the service on the server.  The same goes for all the UUIDs that show up in these filters.  We do have the translated for periods because they are found in [lib/period.js](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/lib/period.js#L15).

This PR is a first step to implementing what @lomamech just finished doing on the client - having human readable translation text for the filters.  It will require an extension of the `FilterService` on the client first, so that the `displayValue` is also back with every filter.  I'll propose a design in an issue.